### ISSUE TITLE
Show size of downloading file (video) before starting download itself.

### ISSUE THREAD

**Issue by heartlog** [libre-tube/LibreTube#3278]
# `Feature request`
## Show size of file after choosing the resolution and audio quality.
1.After selecting the resolution of the file, the application could calculate the size of the file to be downloaded.

2.The calculated size of the file could then be displayed next to the resolution option selected, perhaps in a different color or font to make it more visible.

# How should it be ?
### Selection of video resolution
![Screenshot_20230310_132832_LibreTube](https://user-images.githubusercontent.com/76758776/224265066-c7710030-3d55-4324-970b-2302d75808a5.jpg)

### Select audio quality
![Screenshot_20230310_132839_LibreTube](https://user-images.githubusercontent.com/76758776/224265077-723aad9f-d13c-4692-9cbc-804538a3d681.jpg)

### Resulting last interface
![Screenshot_20230310_134103_LibreTube](https://user-images.githubusercontent.com/76758776/224265083-c687ca9c-a9ec-4b07-996e-a11e470852f4.jpg)

## Why we need it ?
By this user's can download video according to storage and dataplan while considering the network 🛜 speed..
Not to mention Newpipe already have this feature, you can see that for reference.. 

# 👍

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
How would you want to calculate the file sizes of the download? The API doesn't return information about it, therefore it could only be guessed based on a formula. Fetching the size for each possible stream individually would cause a lot of unneeded network traffic.
I found an [article](https://www.circlehd.com/blog/how-to-calculate-video-file-size) which could be started of with, however I'm not sure how appropriate it would be.

**Comment by heartlog** [comment]
Well I have an idea
Well you you can try [Stackoverflow article](https://stackoverflow.com/questions/54034341/how-to-get-size-of-the-video-file-from-url)

**Comment by Bnyro** [comment]
I already stated out previously that it's not a solution as it would cause unneeded network traffic.

**Comment by heartlog** [comment]
Why not try then newpipes method ?

**Comment by heartlog** [comment]
As far as i khow newpipe also extract the video source url and somehow determines the size of video

**Comment by heartlog** [comment]
size of a video in Java based on its playback URL, you can use following code ..

```java
import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;

public class VideoSize {

    public static void main(String[] args) throws IOException {
        URL url = new URL("https://pipedproxy-bom.kavin.rocks/videoplayback?expire=1678474107&ei=GycLZPqaAcyGvw%3D%3D&cpn=jbveMn4EKPCy0CDH&host=rr1---sn-cvh76nls.googlevideo.com");

        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("HEAD");
        int fileSize = connection.getContentLength();
        System.out.println("File size: " + fileSize + " bytes");
    }
}

```
This code sends a HEAD request to the video URL to retrieve the header information, which includes the content length of the video file. The content length is then printed to the console in bytes.

**Comment by FireMasterK** [comment]
You can send a HEAD request to each streaming URL, then calculate it based on the `Content-Length` header.

**Comment by Bnyro** [comment]
I'm aware, we're already doing that when starting a download. However, that would need to be done for 20 streams in parallel and in my opinion cause unneeded server traffic.

**Comment by heartlog** [comment]
> I'm aware, we're already doing that when starting a download. However, that would need to be done for 20 streams in parallel and in my opinion cause unneeded server traffic.

How about using nested list like structure?
Ex.
1) 144p ↓
 -      MP4
 -      Webp
 -      Bruh
2) 240p↓
 -      MP4
 -      Webp
 -      Bruh
3) 360p ↓
 -      MP4
 -      Webp
 -      Bruh
4) 480p ↓
 -      MP4
 -      Webp
 -      Bruh

Then if I click 480 then the format option will become apparent. Like this you can divide 20 stream according and limit less request to server at time.

**Comment by heartlog** [comment]
@Bnyro 
### This is my idea
![Untitled43_20230311155102](https://user-images.githubusercontent.com/76758776/224479076-3a10dc77-27a1-4c8e-aa07-a00ecfcb30c4.png)

**Comment by heartlog** [comment]
@Bnyro can you check this ? It might help [this](https://github.com/libre-tube/LibreTube/issues/3278#issuecomment-1464880030)

**Comment by FireMasterK** [comment]
https://github.com/libre-tube/LibreTube/blob/master/CODE_OF_CONDUCT.md

Please don't tag maintainers/members.

**Comment by heartlog** [comment]
Sorry, about that

**Comment by Bnyro** [comment]
That's now possible thanks to https://github.com/TeamPiped/Piped-Backend/pull/599

**Comment by heartlog** [comment]
Thank God !!
Good work Bnyro !! 🥰
